### PR TITLE
[S5 39] Added Product PATCH and DELETE 

### PR DESCRIPTION
**Pull request by volungeviciute**
Added PATCH and DELETE for the owner side

### CODE REVIEW

File: src/FlowerShop.Core/Entities/Product.cs [hunk 1]
```diff
@@ -24,5 +24,28 @@ public Product(string name, decimal price, string description, ProductType produ
...
         public void SetAvailabilityCount(int newCount) => AvailabilityCount = newCount;
+
+        public void UpdateProductDetails(string name, decimal? price, string description, int? quantity)
+        {
+            if(name != null)
+            {
+                Name = name;
+            }
+
+            if(price != null)
```
Comment (Pilkas-git) [inline src/FlowerShop.Core/Entities/Product.cs]: Nullables has a HasValue property.

File: src/FlowerShop.Core/Entities/Product.cs [hunk 2]
```diff
@@ -24,5 +24,28 @@ public Product(string name, decimal price, string description, ProductType produ
...
+            {
+                Price = (decimal)price;
```
Comment (Pilkas-git) [inline src/FlowerShop.Core/Entities/Product.cs:37]: nullable has GetValueOrDefault().  Price = price.GetValueOrDefault() and you can skip the null check as it assigns a default value if null or a normal value if it's not null.

**Review by Pilkas-git — APPROVED** [review]
Minor optional changes.

File: src/FlowerShop.Core/Entities/Product.cs (hunk 2 above)
Comment (volungeviciute) [reply src/FlowerShop.Core/Entities/Product.cs:37]: GetValueOrDefault  sets value to 0 if the price is null >;)

Comment (Pilkas-git) [reply src/FlowerShop.Core/Entities/Product.cs:37]: You can pass existing Price to GetValueOrDefault(Price), then the default value will remain as Price. 👀 The way we updated entities by using CurrentValues and SetValues  update only the properties which were changed, so there shouldn't be any performance impact if you change the price to what it was as it won't get detected as a changed entity property.

**Pull request merged**